app.go

```diff
@@ -522,6 +522,13 @@ func (app *app) runShell(s string, args []string, prefix string) {
 		cmd.Stderr = os.Stderr
 
 		app.runCmdSync(cmd, prefix == "!")
+
+		//mark the current directory as updated for refresh
+		app.nav.currDir().updated = true
+		
+		//trigger update after returning from shell
+		app.nav.renew()
```

Comment:
This is already called in `app.runCmdSync`, so you should be able to remove this and move the `updated = true` line into there.

Reply:
Fixed